ioread.cpp

```diff
@@ -812,25 +858,48 @@ template<unsigned long int N> bool readFsGridVariable(
    MPI_Comm_size(MPI_COMM_WORLD, &size);
    MPI_Comm_rank(MPI_COMM_WORLD, &myRank);
 
-   std::array<int32_t,3>& localSize = targetGrid.getLocalSize();
-   std::array<int32_t,3>& localStart = targetGrid.getLocalStart();
-   std::array<int32_t,3>& globalSize = targetGrid.getGlobalSize();
+   std::array<FsGridTools::FsIndex_t,3>& localSize = targetGrid.getLocalSize();
+   std::array<FsGridTools::FsIndex_t,3>& localStart = targetGrid.getLocalStart();
+   std::array<FsGridTools::FsSize_t,3>& globalSize = targetGrid.getGlobalSize();
 
    // Determine our tasks storage size
    size_t storageSize = localSize[0]*localSize[1]*localSize[2];
 
    preparations.stop();
 
-   if(size == numWritingRanks) {
-      // Easy case: same number of tasks => slurp it in.
+   std::array<FsGridTools::Task_t,3> fileDecomposition={0,0,0};
+   // No override given (zero array)
+   if(P::overrideReadFsGridDecomposition == fileDecomposition){
+      // Try and read the decomposition from file
+      if (readFsgridDecomposition(file, fileDecomposition) == false) {
+         exitOnError(false, "(RESTART) Failed to get Fsgrid decomposition", MPI_COMM_WORLD);
+      }
+   } else { 
```

Comment:
So... now that you found the GlobalIDs in the .vlsv files, do we even need this manual override for reading any more?

Reply:
Not really, but it has potential debugging value if something breaks down the line.